### ISSUE TITLE
Request to Rename .SRM back to .SAV Nestopia UE

### ISSUE THREAD

**Issue by shinra358** [libretro/RetroArch#3360]
I would like to request that the .srm save files be named back to .sav for easy travel back and forth to and from the normal emulators. Thanks in advance.

**Comment by fr500** [comment]
This is not gonna happen. The SAVERAM interface uses SRM and it's not implemented on a per-emulator basis

**Comment by Alcaro** [comment]
To expand on that answer:

For some emulators, .sav is the easy one.

For some emulators, .srm is the easy one.

No matter what we do, someone's gonna be unhappy.

And any change, in either direction, will force current RetroArch users to rename their SRAMs to keep using RetroArch.

Therefore, we're keeping it this way. While there are benefits to changing, the benefits to staying are a lot bigger.

**Comment by shinra358** [comment]
I know we talked about his before. But here is a different take. Since a simple windows rename from .srm to .sav still makes the saves work, how about an option in retroarch to choose which one you want when it is written to disk and depending on what you chose, it could be changed by retroarch and it could be coded for the cores to still be able to read them? What do you think of that Alcaro and fr500?

'For some emulators, .srm is the easy one.'
except I've seen no emulator that uses these except probably mednafen and retroarch :package: 

'And any change, in either direction, will force current RetroArch users to rename their SRAMs to keep using RetroArch.'
why one or the other when it is code. option in the save category would do justice.

'And any change, in either direction, will force current RetroArch users to rename their SRAMs to keep using RetroArch.'
I could easily make an autohotkey script for batch filename changing and it would take no time for me at all to make or distribute and it'll take no time at all in the conversion process when it is used.

'The SAVERAM interface uses SRM and it's not implemented on a per-emulator basis'
so does this mean it is based on the individual cores wether either is used or not?

**Comment by fr500** [comment]
I still think no.
The autohotkey thing doesn{t solve anything. We're not only on windows.

> except I've seen no emulator that uses these except probably mednafen and retroarch : P
Snes9x?

This is pointless really.
Why don't you tell mednafen dev to change their format instead? or offer the choice? 

> so does this mean it is based on the individual cores wether either is used or not?
what? it's implemented in the core but the actual saving is performed by the frontend

**Comment by shinra358** [comment]
dude, chill w/ the hostility. I'm just asking a question to learn in the nicest way possible.

 yes, that is what I asked. Whether or not it is based on the individual cores.
So by you saying to ask mednafen, even though I'm not even talking about mednafen, does that mean that it is based on how the individual cores were coded? A yes or no would suffice.

**Comment by fr500** [comment]
I don't see any hostlily in my words at all.

You question doesn't make sense to me. I already told you, the cores
implement the saving interface,  the front-end handles writing.  So the
front-end picks the extension for all the cores that implement said
interface.

**Comment by Alcaro** [comment]
> except I've seen no emulator that uses these except probably mednafen and retroarch : P

All SNES emulators I've seen use SRM.

RetroArch was born from a bsnes frontend.

Therefore, SRM.

> Whether or not it is based on the individual cores.

The cores give the frontend a blob of memory, and its size. There are no filenames, the frontend makes one up.

> So by you saying to ask mednafen, even though I'm not even talking about mednafen, does that mean that it is based on how the individual cores were coded? A yes or no would suffice.

You're clearly talking about at least one standalone emulator, so we're assuming it's Mednafen until you tell us what it actually is.

Why should we be the one to change? Why not tell Mednafen to use SRM instead?

I personally have no direct objections towards a SRM name switch, but I don't think it's worth the effort.

**Comment by shinra358** [comment]
Just responding to your responses. Not prolonging the convo:


"The cores give the frontend a blob of memory, and its size. There are no filenames, the frontend makes one up."
I don't understand what you are saying here. This part that you quoted was me talking about whether or not the save types are chosen per core or not. I don't understand your answer here in relationship to that. Let me clarify further:
Is it the core's job in determining what save type it will use? By how you are speaking, I can't tell if you're saying it's the frontend's job or the core's job.


'You're clearly talking about at least one standalone emulator, so we're assuming it's Mednafen until you tell us what it actually is.'
I'm asking about ANY emu in relation to your statement. When one doesn't specific a type of thing, it usually does mean all in the category spoken about. Which are only 3 emulators we've all named here + a couple of snes emus so far out of the many others. The only thing I use mednafen for is for virtualboy only.

'Why should we be the one to change? Why not tell Mednafen to use SRM instead?'
Asking about an option to request a user to determine either/or is not telling you to change. It's asking could we have an option to decide for ourselves because of the earlier reasons. Mednafen is not the only emulator in existance. Out of all 2 or 3 emus(+2snes emus) that use srm, the other 50 use savs. *shrugs*


Again, I'm only responding to what you asked at this point. No need (referring to anyone) to come back at me as if I stole something with hostility and insults. Just a convo. Which by now is practically over.

**Comment by fr500** [comment]
Again with the "hostility" argument.
I was never hostile, you don't really want to understand the responses you're getting.

Anyway, this is getting nowhere.

**Comment by shinra358** [comment]
'This is pointless really.'
hostile.

'I was never hostile, you don't really want to understand the responses you're getting, that's all.'
that's why I asked the question that you were skipping around...

'but you can't expect the devs to accomodate every request'
wasn't demanding anything.

'fr500 locked and limited conversation to collaborators '
rage.

^hostility and rage...


Just look how Alcaro responds and compair to how you respond. The last simple question I asked (for the 3rd time) was that I didnt understand whether or not you meant the frontend was responsible for the conversions or the cores. That's it. I asked you for a yes or a no. Yet you responded as if I'm bugging you so badly and just kept talking and comparing as you are offending by a normal conversation. When I mentioned hostility again to Alcaro, I was talking about you because from the start, it looks like you are furious. I specifically said I was just responding to Alcaro's comments at this point. Yet you come back again prolonging an argument when I already stated that was the end and that I was trying to get an answer to that one question because of your rage and skipping over the question every time.

Not once have I said anything mean or hostile for you to talk that way. The convo was over at your post of 'I still think no'. After that, I was only trying to find out what you meant.

'Again, I'm only responding to what you asked at this point. No need (referring to anyone) to come back at me as if I stole something with hostility and insults. Just a convo. Which by now is practically over.'

That comment meant you and I wanted to make sure to clarify that I'm not even asking for it anymore and am just trying to figure out your answer. Yet, you come back raging anyway.
Peace. If you're still on your frame of mind, then then I don't know what to tell ya. If you're not really a native english speaker, I can understand. So if you are, that is what hostility sounds like. I got str8 A's in english classes so I know. I'm also 33 with kids. So I know what tones of speech is.


' The SAVERAM interface uses SRM and it's not implemented on a per-emulator basis'
This, and this:
' I already told you, the cores implement the saving interface, the front-end handles writing.'

is not the same. And this:
'The cores give the frontend a blob of memory, and its size. There are no filenames, the frontend makes one up.'
just confuses what I thought the first 2 meant.

All you had to say in the beginning (when I asked the question) was:
The cores implement the saving interface, the front-end handles writing. Which is what I was asking. When you made this post that answered the question, again, it was OVER. Whatever came after is me responding to Alcaro as I still already stated. And then after that, you came back still saying the same thing you said before when the convo was over elongating the convo more and telling me I'm elongating the convo.

So if you don't know, the convo is over, the question has been answered. And therefore this is the end. Got it? Got it. So since this is the end now, if you come back telling me how stupid I am or something of the like, that would be you being hostile and making things longer. So, I'm done. Okay? It's all over. I need no more responses. Kapeesh? Kapeesh.

**Comment by fr500** [comment]
So stating that something pointless is pointless is hostility now.  Ok.

Guess we're back in elementary school.
Keep reading between lines all that your want.

**Comment by shinra358** [comment]
Exactly the thing I said you would do, you do it.
G'nite dude. If I talk to you like you are to me right now, your finger would be over the ban button.
Bye.

**Comment by fr500** [comment]
No, I didn't call you stupid, I was never hostile, and there is no ban button. We never really ban anyone anyhow.

All the answers provided say the same thing. I elaborated in my second attempt after you didn't understand. I could be more explicit yes but other than that I see no issues with my responses.

I never said that YOU were demanding either, I was talking from former experience.

You just joined and I have had 26 notifications from you. You have made our issue tracker a public forum.
I understand you want to leave on a "high note" but you're not getting that.

Have fun.
